tox.ini

```diff
@@ -1,48 +1,167 @@
+passenv=
+    TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH TEST_DATABASE_URL TEST_RUNNER
+setenv=
+    DJANGO_SETTINGS_MODULE=teachers_digital_platform.tests.settings
+    PYTHONPATH={toxinidir}/cfgov-refresh/cfgov:{env:PYTHONPATH:}
+    TEST_DATABASE_URL=postgres://cfpb@localhost/cfgov
+deps=
+    current: {[current-config]deps}
 commands=
-    lint: {[lint]commands}
     ./install_cfgov_refresh.sh
     coverage erase
     coverage run --source='teachers_digital_platform' {envbindir}/django-admin.py test {posargs}
     coverage report -m
-setenv=
-    DJANGO_SETTINGS_MODULE=teachers_digital_platform.tests.settings
-    PYTHONPATH={toxinidir}/cfgov-refresh/cfgov:{env:PYTHONPATH:}
 
-basepython=
-    py36: python3.6
 
+[current-config]
+# Configuration necessary for the production versions Python, Django and 
+# Wagtail
+# Note: This is not an env and will not run if invoked. Use:
+# 
+#   tox -e unittest-current
+#   tox -e unittest
+# 
+# To run unittests with the current production versions.
+basepython=python3.6
 deps=
-    dj111: Django>=1.11,<1.12
-    dj20: Django>=2.0,<2.1
-    dj22: Django>=2.2,<2.3
-    wag23: wagtail>=2.3,<2.4
-    wag28: wagtail>=2.8,<2.9
-    lint:  {[lint]deps}
-
-[lint]
+    Django>=1.11,<1.12
+    wagtail>=2.3,<2.4
+
+
+[future-config]
+# Configuration necessary for upcoming versions of Python, Django, and 
+# Wagtail. This is the place where the "future" dependencies will need to be 
+# modified when we're ready to track new versions of Python, Django, or 
+# Wagtail.
+# Note: This is not an env and will not run if invoked. Use:
+# 
+#   tox -e unittest-future
+# 
+# To run unittests with the upcoming versions specified here.
+basepython=python3.6
 deps=
-    flake8>=2.2.0
-    isort>=4.2.15
-commands=
-    flake8 .
-    isort --check-only --diff --recursive teachers_digital_platform
+    Django>=2.0,<2.1
+    wagtail>=2.8,<2.9
+
+
+[testenv:unittest-future]
+# Invoke with: tox -e unittest-future
+recreate=False
+basepython={[future-config]basepython}
+envdir={toxworkdir}/unittest-future
+deps=
+    {[future-config]deps}
+commands={[unittest-config]commands}
+
+
+[testenv:unittest]
+# Invoke with: tox -e unittest
+# This should run identically to tox -e unittest-current
+recreate=False
+basepython={[current-config]basepython}
+envdir={toxworkdir}/unittest-current
+deps=
+    {[current-config]deps}
+    {[unittest-config]deps}
+commands={[unittest-config]commands}
+
+
+[testenv:lint]
+# Invoke with: tox -e lint
+# This should run identically to tox -e lint-current
+recreate=False
+basepython={[current-config]basepython}
+envdir={toxworkdir}/lint-current
+deps={[lint-config]deps}
+commands={[lint-config]commands}
 
 [flake8]
-ignore = 
-    # Allow assigning lambda expressions
-    E731,
-    # Allow line breaks after binary operators
-    W503,
-
-exclude =
-    # Some of this are excluded for speed of directory traversal. Not all of 
-    # them have Python files we wish to ignore.
```

Comment:
I think for reference, I'd like to see the comments remain.

Reply:
Added comments